pkg/server/auth/server.go

```diff
@@ -464,12 +464,10 @@ func (s *AuthServer) UserInfo(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	info, err := s.provider.UserInfo(r.Context(), oauth2.StaticTokenSource(&oauth2.Token{
-		AccessToken: c.Value,
-	}))
+	info, err := s.verifier().Verify(r.Context(), c.Value)
```

Comment:
c.Value is the access_token rather than the id_token?

Reply:
Not necessarily...look at findToken.

There's an argument to be made that findToken should not return the access_token at all

And a further argument that if we can't use the ID token, we should fall back to the access token, but...it's pretty bad if we can't use the ID token?

Reply:
Yeah. I guess accesstoken was just to query user info? We can drop it now maybe?

Findtoken only returns access token if it's present tho? 👀